### PR TITLE
OIDC: update sync3 tables

### PR DESCRIPTION
**Pull request by DMRobertson**
Part 2 of N, per https://github.com/matrix-org/sliding-sync/issues/89#issuecomment-1527909808.

This changes the remaining database tables so that any per-device data is associated with a (user_id, device_id) pair.

The next step will be to write migration logic.

### CODE REVIEW

File: state/to_device_table.go [hunk 1]
```diff
@@ -65,29 +69,31 @@ func NewToDeviceTable(db *sqlx.DB) *ToDeviceTable {
 	return &ToDeviceTable{db}
 }
 
-func (t *ToDeviceTable) SetUnackedPosition(deviceID string, pos int64) error {
-	_, err := t.db.Exec(`INSERT INTO syncv3_to_device_ack_pos(device_id, unack_pos) VALUES($1,$2) ON CONFLICT (device_id)
-	DO UPDATE SET unack_pos=$2`, deviceID, pos)
+func (t *ToDeviceTable) SetUnackedPosition(userID, deviceID string, pos int64) error {
+	_, err := t.db.Exec(`INSERT INTO syncv3_to_device_ack_pos(user_id, device_id, unack_pos) VALUES($1,$2,$3) ON CONFLICT (user_id, device_id)
+	DO UPDATE SET unack_pos=excluded.unack_pos`, userID, deviceID, pos)
```
Comment (DMRobertson) [inline state/to_device_table.go:74]: Minor thing, but I nearly missed that `$2` was used twice!

**Review by kegsay — APPROVED** [review]
Missing device_data table?

File: state/to_device_table.go [hunk 2]
```diff
@@ -55,7 +57,9 @@ func NewToDeviceTable(db *sqlx.DB) *ToDeviceTable {
 		action SMALLINT DEFAULT 0 -- 0 means unknown
 	);
 	CREATE TABLE IF NOT EXISTS syncv3_to_device_ack_pos (
-		device_id TEXT NOT NULL PRIMARY KEY,
+	    user_id TEXT NOT NULL,
+		device_id TEXT NOT NULL,
```
[the remark this answers is not in the document]
Comment (DMRobertson) [reply state/to_device_table.go:61]: It's a tabs versus spaces thing. I guess gofmt didn't touch this as it's inside a string literal.

**Comment by DMRobertson** [comment]
> Missing device_data table?

See https://github.com/matrix-org/sliding-sync/pull/95#issuecomment-1531870459

**Pull request merged**